### PR TITLE
Integrate the PAKE Extension into the Crypto API specification

### LINKED ISSUE #170: Integrate the PAKE Extension into the Crypto API specification
Now that the PAKE Extension is Final, and is subject to the same future compatibility requirements as the rest of the Crypto API, it should be incorporated into the primary specification.

### PR DESCRIPTION
**Pull request by athoelke**
_[Updated: rebased the PR after publication of Crypto API 1.2.1, and updated the rendered PDF]_

As an initial step for the Crypto API 1.3 release, integrate the API for PAKE operations into the primary specification, now that the PAKE Extension is Final.

Fixes #170.

A draft PDF render of the integrated specification: [IHI0086-PSA_Certified_Crypto_API-1.3.0-integrate-pake-draft.2.pdf](https://github.com/ARM-software/psa-api/files/14790999/IHI0086-PSA_Certified_Crypto_API-1.3.0-integrate-pake-draft.2.pdf)

### Notes on the integration

* Most of the extension forms a new sub-chapter within Cryptographic operations, but the key type and encoding information for SPAKE2+ is integrated into the respective sections.
* Previous indirect references to the Crypto API, have been replaced with direct cross-references. (I hope I didn't miss any)
* I have reworked parts of the Functionality chapter, as multi-part operations are now used with asymmetric cryptography as well.
* Lists of 'key creation functions' all include the `psa_pake_get_shared_key()`.

### CODE REVIEW

File: doc/crypto/overview/functionality.rst [hunk 1]
```diff
@@ -134,28 +135,16 @@ Most implementations provide the following functions:
...
-*   Key derivation. See :secref:`kdf`.
-
-For each type of symmetric cryptographic operation, the API can include:
-
-*   A pair of *single-part* functions. For example, compute and verify, or
-    encrypt and decrypt.
-*   A series of functions that permit *multi-part operations*.
-
-Key derivation only provides multi-part operation, to support the flexibility required by these type of algorithms.
+*   One or more *single-part* functions, that carry out one function of that operation. For example, compute, verify, encrypt, or decrypt.
```
Comment (davidhorstmann-arm) [inline doc/crypto/overview/functionality.rst:143]: Nit / editorial: I found this wording slightly confusing. Would something like:

```suggestion
*   One or more *single-part* functions, which perform a whole operation in a single function call. For example, compute, verify, encrypt or decrypt.
```
Work as an alternative?

**Review by davidhorstmann-arm — CHANGES_REQUESTED** [review]
One nit / wording comment on the functionality section.

I think I also found a [preexisting copypasta](https://github.com/athoelke/psa-api/blob/crypto-1.3-integrate-pake/doc/crypto/api/ops/pake.rst?plain=1#L594) in the PAKE spec:

```rst
    *   Assign the result of the function `psa_pake_cipher_suite_init()` to the object, for example:

        .. code-block:: xref

            psa_pake_operation_t operation;
            operation = psa_pake_operation_init();
```
I believe the `psa_pake_cipher_suite_init()` should be `psa_pake_operation_init()`.

If it's out of scope for this PR I can raise a followup.

Otherwise, the code motion looks good and `psa_pake_get_shared_key()` seems to be listed everywhere that it should.

**Comment by athoelke** [comment]
> I think I also found a [preexisting copypasta](https://github.com/athoelke/psa-api/blob/crypto-1.3-integrate-pake/doc/crypto/api/ops/pake.rst?plain=1#L594) in the PAKE spec:

Well spotted - I have added a commit for this. This is a good PR to fix that error.

File: doc/crypto/overview/functionality.rst [hunk 2]
```diff
@@ -168,7 +157,7 @@ use of a `multi-part operation <multi-part-operations>`.
 .. _multi-part-operations:
 
 Multi-part operations
-^^^^^^^^^^^^^^^^^^^^^
+~~~~~~~~~~~~~~~~~~~~~
 
 Multi-part operations are APIs which split a single cryptographic operation into
 a sequence of separate steps. This enables fine control over the configuration
```
[the remark this answers is not in the document]
Comment (athoelke) [reply doc/crypto/overview/functionality.rst:163]: I've updated the text along these lines, but keeping consistency with the wording of the setup functions.

I also took the opportunity to add PAKE as an example of why multi-part operations are needed.

**Pull request merged**